internal/cmd/check.go

```diff
@@ -0,0 +1,73 @@
+package cmd
+
+import (
+	"github.com/sonatype-nexus-community/nancy/buildversion"
+	"github.com/sonatype-nexus-community/nancy/settings"
+	"github.com/sonatype-nexus-community/nancy/update"
+	"time"
+)
+
+// For use in checking for newer release version during app startup (not during explicit command to check for update)
+func checkForUpdates(gitHubAPI string) error {
+	updateCheck := &settings.UpdateCheck{
+		LastUpdateCheck: time.Time{},
+	}
+
+	err := updateCheck.Load()
+	if err != nil {
+		return err
+	}
+
+	if update.ShouldCheckForUpdates(updateCheck) {
```

Comment:
This is probably fine....but thinking of how we do CI.... not much gets cached. So once we make this automatic its going to be checking all the time if the underlying docker CI containers always get blown away. We we still want to do it automatically we probably need to add something to the README about caching the ~/.ossindex dir.

Reply:
> This is probably fine....but thinking of how we do CI.... not much gets cached. So once we make this automatic its going to be checking all the time if the underlying docker CI containers always get blown away. We we still want to do it automatically we probably need to add something to the README about caching the ~/.ossindex dir.

Great point! I'll try to add some doco about this. I actually like the idea of it printing messages about an updated Nancy version in CI builds. Slightly increasing the chance a human will notice it in the CI logs. Still worth mentioning though.

Reply:
Added some docs here: c62fded